This replaces some conditional template haskell by static code. Replace by the
other variant we we start shipping Cabal-1.17


Index: haskell-llvm-base-3.2.0.2/Setup.hs
===================================================================
--- haskell-llvm-base-3.2.0.2.orig/Setup.hs	2013-05-28 02:38:14.000000000 +0200
+++ haskell-llvm-base-3.2.0.2/Setup.hs	2014-03-29 17:19:03.463550263 +0100
@@ -1,7 +1,5 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE PatternGuards #-}
-{-# LANGUAGE TemplateHaskell #-}
-{-# LANGUAGE QuasiQuotes #-}
 import System.Directory
 import System.Environment
 import System.FilePath
@@ -19,7 +17,6 @@
 import Distribution.Simple.Register
 import Distribution.Simple.Utils
 import Distribution.Text ( display )
-import Language.Haskell.TH

 main = do
     let hooks = autoconfUserHooks { postConf = if os == "mingw32"
@@ -103,7 +100,8 @@
 -}

 --- horrible hack to support cabal versions both above and below 1.17
-extractCLBI x=
+extractCLBI x = let LocalBuildInfo { libraryConfig  = Just clbi } = x in clbi
+{-
     $(if cabalVersion >= Version [1,17,0] []
         then  appE (appE  ( varE $ mkName "getComponentLocalBuildInfo") ( varE 'x) ) (conE ( mkName "CLibName"))

@@ -114,6 +112,7 @@
                              (conP (mkName "Just")    [varP $ mkName "clbi"] ) ] )
                     (normalB $ varE 'x)   []    ]
                  (varE $ mkName "clbi")  )
+-}

 register' :: PackageDescription -> LocalBuildInfo
           -> RegisterFlags -- ^Install in the user's database?; verbose
